samples/drivers/drv2605/src/main.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2024 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/haptics.h>
+#include <zephyr/drivers/haptics/drv2605.h>
```

Comment:
Since the configuration of the haptic device is vendor/device specific, it is more efficient to move the/all configuration functions to the vendor/device specific header :)

Specifically, the API `haptics_source_config(dev, HAPTICS_SOURCE_ROM, &rom_data);` is not portable, whereas `haptics_start_output(dev);` is.

Forcing the config struct, in this case `struct drv2605_rom_data rom_data = {};` to be passed by void reference through a "generic" API is less safe than creating a specific API for the drv2605 which takes a type safe combination of arguments, like:

```
int haptics_source_config_rom(const struct device *dev, const struct drv2605_rom_data *data);
```
Which is completely type safe (except for the *dev pointer but that's a Zephyr thing).

In summary, device specific configs/features which are not generic enough to be abstractly used by the application (one can switch from one haptic device of one vendor to another, without changing the application), write device specific typesafe APIs for it. Only expose the generic APIs in the haptics API

Reply:
ack, I see your point here.